src/mokernels/independent.jl

```diff
@@ -23,20 +23,39 @@ struct IndependentMOKernel{Tkernel<:Kernel} <: MOKernel
     kernel::Tkernel
 end
 
+# kernel function should be symmetric
+# would really like (κ::IndependentMOKernel)((x, px)::Tuple{T,Int}, (y, py)::Tuple{T,Int}) where T, but seems to cause autodiff problems
 function (κ::IndependentMOKernel)((x, px)::Tuple{Any,Int}, (y, py)::Tuple{Any,Int})
     if px == py
         return κ.kernel(x, y)
     else
-        return 0.0
+        retType = Base.return_types(κ.kernel, (typeof(x), typeof(y)))[1]
```

Comment:
```suggestion
        retType, = Base.return_types(κ.kernel, (typeof(x), typeof(y)))
```
encodes in code that we expect the kernel to return a single value, makes it easier to read & also errors instead of just a silent bug in case kernel() happens to return more than one value..

Reply:
I did not know about this syntax, thank you.

Reply:
:) that's what I love about PR reviews, learning how to write better code!
and I learned that this is actually all superseded by what @willtebbutt wrote below ;)